src/components/oculus-touch-controls.js

```diff
@@ -320,20 +357,130 @@ module.exports.Component = registerComponent('oculus-touch-controls', {
     });
   },
 
+  applyOffset: function (model) {
+    model.position.copy(this.displayModel[this.data.hand].modelPivotOffset);
+    model.rotation.copy(this.displayModel[this.data.hand].modelPivotRotation);
+  },
+
+  multiMeshFix: function (button) {
+    if (!this.buttonMeshes[button]) return;
+    this.buttonMeshes[button].traverse((node) => {
+      if (node.type !== 'Mesh') return;
+      var originalMaterial = node.material;
+      this.buttonMeshes[button].naturalColor = originalMaterial.color;
+      node.material = new THREE.MeshStandardMaterial();
+
+      // preserve details; not iterable, so we have to use Object.keys()
+      Object.keys(originalMaterial).forEach(key => {
+        if (originalMaterial[key] !== node.material[key]) {
+          // because A-Frame uses the same single material object for all material nodes, it also uses the
+          // same color. This means they use the same color object--we have to break that link. Other links
+          // likely also exist, but they don't matter to us, because color is the only one we change.
+          node.material[key] = key !== 'color' ? originalMaterial[key] : originalMaterial[key].clone();
+        }
+      });
+      originalMaterial.dispose();
+    });
```

Comment:
So in the end you just want to clone the material, using the same textures, but have a different color instance, right?
Material has a copy method that do just this, so you can use it and just clone the color property afterwards. This should work:
```
node.material = new THREE.MeshStandardMaterial();
node.material.copy(originalMaterial);
node.material.color = originalMaterial.color.clone();
originalMaterial.dispose();
```

Reply:
I opted for a variation on this approach:
```js
      let newMaterial = node.material.clone();
      this.buttonMeshes[button].naturalColor = node.material.color;
      node.material.dispose();
      node.material = newMaterial;
```

No special handling is needed for color now, since we're calling `clone()` it seems to recursivley clone all the child elements as well, I imagine / it seems. Again, probably slight efficiency reduction, but definitely better code.